internal/service/pushbasedlogexport/resource.go

```diff
@@ -51,7 +52,15 @@ func (r *pushBasedLogExportRS) Create(ctx context.Context, req resource.CreateRe
 	connV2 := r.Client.AtlasV2
 	projectID := tfPlan.ProjectID.ValueString()
 	if _, err := connV2.PushBasedLogExportApi.CreatePushBasedLogConfiguration(ctx, projectID, logExportConfigReq).Execute(); err != nil {
-		resp.Diagnostics.AddError("eError when creating push-based log export configuration", err.Error())
+		resp.Diagnostics.AddError("Error when creating push-based log export configuration", err.Error())
+
+		log.Printf("[INFO] Unconfiguring push-based log export for project due to create failure: %s", projectID)
+		if _, err := connV2.PushBasedLogExportApi.DeletePushBasedLogConfiguration(ctx, projectID).Execute(); err != nil {
```

Comment:
just curious, in what scenario does the create fail but we still have to follow up with a delete of the resource?

Reply:
> wouldn't the delete affect some other configuration that was perhaps created elsewhere?

We could limit this follow-up delete operation to only be called if the create operation resulted in `BUCKET_VERIFICATION_FAILED` or `ASSUME_ROLE_FAILED`, providing this information to the user. This way if creation fails due to these cases, the create operation will bring the configuration back to `UNCONFIGURED` (currently we fail but the configuration is also left in a failed state).

Reply:
added